builder/simplify.cc

```diff
@@ -461,6 +514,7 @@ class simplifier : public node_mutator {
           result = crop_dim::make(std::get<0>(*i), std::get<1>(*i), std::get<2>(*i), result);
         }
         set_result(mutate(result));
+        merge_loop_references(std::move(old_references));
```

Comment:
We should merge the references here, but we shouldn't mark the references as being in a loop.